staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/validation/validation.go

```diff
@@ -42,5 +42,36 @@ func ValidateListOptions(options *internalversion.ListOptions) field.ErrorList {
 			allErrs = append(allErrs, field.Forbidden(field.NewPath("resourceVersionMatch"), "resourceVersionMatch \"exact\" is forbidden for resourceVersion \"0\""))
 		}
 	}
+	if options.SendInitialEvents {
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("sendInitialEvents"), "sendInitialEvents is forbidden for list"))
+	}
+	return allErrs
+}
+
+func validateWatchOptions(options *internalversion.ListOptions, isWatchListFeatureEnabled bool) field.ErrorList {
```

Comment:
I think I got slightly lost when reviewing this function, so I need an exact table of what we allow
[I'm assuming it's for watch]

- match=""  -> OK    (backward compatibility, typical old case)
- match=NotOlderThan,  SendInitialEvents  -> OK (new case)

For the case of match="" and  RV=0/RV="", SendInitialEvent will get propagated via default and that will change match to NotOlderThan too, so we will be fine.

Is there anything else we allow? I think that's all.
If so, I think that validation should better reflect that.

```
if len(match) == 0 {
   // typical watch 
  if options.SendInitialEvents {
    forbidden
  }
}
if len(match) > 0 {
  // the new watch with SendInitialEvents
  if match != NotOlderThan {
    forbidden
  }
  if !SendInitialEvents {
   forbidden
  }
  if !FG enabled {
   forbidden
  }
}

if len(continue) > 0 {
  forbidden for watch
}
```

That would be much cleaner for me - WDYT?